src/FirebaseAuth.js

```diff
@@ -222,24 +222,23 @@
      */
     _routerMethodOnAuthPromise: function(rejectIfAuthDataIsNull) {
       var ref = this._ref;
-      var deferred = this._q.defer();
 
-      function callback(authData) {
-        if (authData !== null) {
-          deferred.resolve(authData);
-        } else if (rejectIfAuthDataIsNull) {
-          deferred.reject("AUTH_REQUIRED");
-        } else {
-          deferred.resolve(null);
+      return this._utils.promise(function(resolve,reject){
+        function callback(authData) {
+          if (authData !== null) {
+            resolve(authData);
```

Comment:
No, I have more pedantic reasons for wanting the `return` in there. I'm not worried about the promise getting resolved or rejected multiple times since, as you said, the A++ spec prevents that. But I do think in most cases it is unexpected and unnecessary to run more code after the promise is resolved or rejected. Putting an explicit `return` in there will stop future lines in that function from running, which is what we want in most cases. Not doing so can lead to some nasty bugs where code runs when don't expect it to.

But yes, allowing for chaining is also a positive, so let's do it! But when you do add the `return` statements, you'll need to move the `ref.offAuth(callback)` above the `if` block or else the `return` statements will exit early causing that code to not run.